### PR TITLE
add azuredb integration test

### PR DESCRIPTION
**Pull request by bferguson3**
api/program.cs is now its own class

the internal sql methods need to be factored out and expanded. just for testing.

### CODE REVIEW

**Comment by bferguson3** [comment]
@jlbeard84

File: api/Models/Developer.cs [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+        public string Name { get; set; } = string.Empty;
+        public string? RomanizedName { get; set; } = string.Empty;
+
+        public Employee[]? Credits { get; set; }
+
+        public DateTime? YearsActiveStart { get; set; }
+        public DateTime? YearsActiveEnd { get; set; }
+
+        public Game[]? GamesCreated { get; set; }
```
Comment (jlbeard84) [inline api/Models/Developer.cs:16]: I would change this array types to be List<Game> (or IEnumeraable<Game>) as it buys you a little more with stuff like null bounds checking out of the box.
```suggestion
        public List<Game> GamesCreated { get; set; } = new List<Game>();
```

File: api/Models/Employee.cs [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+    public class Employee 
+    {
+        public string Id { get; set; } = string.Empty;
+
+        public string Name { get; set; } = string.Empty;
+        public string? RomanizedName { get; set; } = string.Empty;
+
+        public Developer? WorksFor { get; set; }
+
+        public Game[]? GamesWorkedOn { get; set; }
```
Comment (jlbeard84) [inline api/Models/Employee.cs:12]: Same here = 
```suggestion
        public List<Game> GamesCreated { get; set; } = new List<Game>();
```

File: api/Models/Game.cs [hunk 3]
```diff
@@ -5,8 +5,11 @@ public class Game
         public string Id { get; set; } = string.Empty;
         
         public string Title { get; set; } = string.Empty;
+        public string? JapaneseTitle { get; set; } = string.Empty;
+        public string? RomanizedTitle { get; set; } = string.Empty;
 
         public DateTime? OriginalReleaseDate { get; set; }
+        public DateTime? MostRecentReleaseDate { get; set; }
```
Comment (jlbeard84) [inline api/Models/Game.cs:12]: Maybe we could move this into a child object for the release, where it's like
```
Release
{
   Platform
   ReleaseDate
   Regions
}
```

File: api/Program.cs [hunk 4]
```diff
@@ -1,30 +1,116 @@
...
+using System.Threading.Tasks;
+using System.Configuration;
+using System.Collections.Generic;
+using System.Net;
+using Microsoft.Azure.Cosmos;
 
-var builder = WebApplication.CreateBuilder(args);
+public class Program
+{
+    private static readonly string EndpointUri = "https://kemcowiki.documents.azure.com:443/";
```
Comment (jlbeard84) [inline api/Program.cs:13]: This stuff is fine here for now - long term we will want to move it into a settings file that we can use when deploying.

File: api/Program.cs [hunk 5]
```diff
@@ -1,30 +1,116 @@
...
+        {
+            Console.WriteLine("End of demo, press any key to exit.");
+            Console.ReadKey();
+        }
+        /////////// Now run API service
+        app.Run();
+    }
 
-app.UseAuthorization();
+    public async Task GetStartedDemoAsync()
```
Comment (jlbeard84) [inline api/Program.cs:71]: Long term we'll slip this into a DB class but if everything works here then the DB should be good to go

File: api/Models/Developer.cs (hunk 1 above)
Comment (bferguson3) [reply api/Models/Developer.cs:16]: thx

File: api/Program.cs (hunk 4 above)
Comment (bferguson3) [reply api/Program.cs:13]: sounds good. was just taken from code examples anyway

File: api/Program.cs (hunk 5 above)
Comment (bferguson3) [reply api/Program.cs:71]: db is good to go but def should be moved out

File: api/Models/Game.cs (hunk 3 above)
Comment (bferguson3) [reply api/Models/Game.cs:12]: Might be good idea, need to discuss further

**Pull request merged**